earthpy/tests/test_plot_draw_legend.py

```diff
@@ -174,6 +174,21 @@ def test_noncont_listed_cmap(binned_array, listed_cmap):
     plt.close(f)
 
 
+def test_classes_provided_as_array(binned_array, listed_cmap):
+    """Test legend fun to ensure classes provided as an array still work."""
+
+    cmap, norm = listed_cmap
+    bins, arr_class = binned_array
+
+    f, ax = plt.subplots(figsize=(5, 5))
+    im = ax.imshow(arr_class, cmap=cmap, norm=norm)
+    leg = ep.draw_legend(im, classes=np.asarray([1, 2, 3, 4, 5]))
```

Comment:
Minor quibble here, but we have the list explicitly written out as `[1, 2, 3, 4, 5]` twice in this test. It might be cleaner to do something like 

```python
    ...
    im = ax.imshow(arr_class, cmap=cmap, norm=norm)
    n_classes = 5
    leg = ep.draw_legend(im, classes=np.arange(n_classes))

    legend_cols = [i.get_facecolor() for i in leg.get_patches()]
    assert len(legend_cols) == n_classes
    plt.close(f)
```

Reply:
ok @mbjoseph  . i had to tweak a few things for that to work given it's 0-based indexing and such! see what you think.

Reply:
Cool - I am noticing that the test still passes even when doing `np.arange(n_classes)`. Is that not supposed to work?

Reply:
it should work. So essentially for a listed colormap you need to provide the number of classes and it will create that number of legend items. They are discrete values so 5 classes = 5 colors. However what should fail is if they provide 6 classes and a listed colormap of 5 colors. in that case the last two values will be the same color.  

the function is tricky because drawing a legend is a manual task in this instance - particularly when some values may be missing and only the user knows what values those are. so i tried to make it very flexible. 

i may add another test for if a use provides 6 classes but there are only 5 colors in the cmap.

Reply:
Ok sounds good @lwasser. I'd advocate for making the tests only as complex as they need to be to test behavior, so in this case using `np.arange(n_classes)` instead of `np.arange(1, 1 + n_classes)` would be the preferred option. 

Let me know when you're ready for another review! (maybe after adding that additional test?)

Reply:
will do. @mbjoseph what do you thijnk about this... so in this test if i rewrite it i could test 2 things.

1. that an array works and 
2. that i get a nice message (i need to add that) if a user provides more classes than colors. 

the code would be the same. Would it be better to repeat the code and have 2 tests OR to have 1 more concise test that hits both things.

Reply:
I would try to shoot for two tests (each with one `assert` statement), since these seem like two different things to test. 

To avoid copypasta, it might be good to think about whether there is a way to use a fixture to share some of the logic among tests. I'm noticing that there already is a fair bit of repetition and shared logic among tests. Maybe some of that can be bundled up in a fixture?

Reply:
ok i'm working on the second test FIRST because it also means adding a test into the function. then i'll think about fixtures! thanks!